whois-api/src/main/java/net/ripe/db/whois/api/rest/HttpsAPIKeyAuthCustomizer.java

```diff
@@ -0,0 +1,128 @@
+package net.ripe.db.whois.api.rest;
+
+import com.google.common.net.HttpHeaders;
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletRequestWrapper;
+import jakarta.servlet.http.HttpServletResponse;
+import net.ripe.db.whois.common.apiKey.ApiKeyAuthServiceClient;
+import net.ripe.db.whois.common.apiKey.ApiKeyUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.jetty.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.Objects;
+
+@Component
+public class HttpsAPIKeyAuthCustomizer implements Filter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HttpsAPIKeyAuthCustomizer.class);
+
+    final private boolean isEnabled;
+    final ApiKeyAuthServiceClient apiKeyAuthServiceClient;
+
+    @Autowired
+    public HttpsAPIKeyAuthCustomizer(@Value("${apikey.authenticate.enabled:false}") final boolean isEnabled,
+                                     final ApiKeyAuthServiceClient apiKeyAuthServiceClient) {
+        this.apiKeyAuthServiceClient = apiKeyAuthServiceClient;
+        this.isEnabled = isEnabled;
+    }
+
+    @Override
+    public void doFilter(final ServletRequest request, final ServletResponse response, final FilterChain chain) throws IOException, ServletException {
+
+        if( !(request instanceof HttpServletRequest)) {
+            chain.doFilter(request, response);
+        }
+
+        final HttpServletRequest httpRequest = (HttpServletRequest) request;
+
+        if(isValidRequest(httpRequest)) {
```

Comment:
Should be named is *NOT* valid request as an error is returned if true? Or else flip the check in the method

Reply:
Right, I will rename